src/mdns/mdns_mdnssd.cpp

```diff
@@ -415,9 +462,133 @@ otbrError PublisherMDnsSd::PublishService(uint16_t aPort, const char *aName, con
     return ret;
 }
 
-Publisher *Publisher::Create(int aFamily, const char *aHost, const char *aDomain, StateHandler aHandler, void *aContext)
+otbrError PublisherMDnsSd::UnpublishService(const char *aName, const char *aType)
+{
+    DiscardService(aName, aType);
+    return OTBR_ERROR_NONE;
+}
+
+otbrError PublisherMDnsSd::PublishHost(const char *aName, const uint8_t *aAddress, uint8_t aAddressLength)
+{
+    otbrError    ret   = OTBR_ERROR_NONE;
+    int          error = 0;
+    char         fullName[kMaxSizeOfDomain];
+    DNSRecordRef hostRecord;
+    Host         newHost;
+
+    SuccessOrExit(ret = MakeFullName(fullName, sizeof(fullName), aName));
```

Comment:
Prefer to use `std::string` so that we won't worry about the memory issues in `strcpy` and `strlen`. Or we should use `strncpy`  and `strnlen` when possible.

Reply:
Yes, life with `std::string` can be much easier. May refactor it in a future PR.